Enumerables

men32z opened the pull request.

men32z committed:
first commit

men32z committed:
my_each

men32z committed:
enumerables until my_map

men32z committed:
my_inject

men32z committed:
my_inject

men32z committed:
all methods finished

men32z committed:
test self redundant

men32z committed:
sticlker errors fixed

men32z committed:
sticlker errors fixed 2

men32z committed:
sticlker errors fixed 3

men32z committed:
sticlker errors fixed 4

men32z committed:
sticlker errors fixed 5

men32z committed:
Update README.md

maelfosso commented on enumerable.rb:
```diff
@@ -0,0 +1,75 @@
+    length.times do |x|
+      yield(self[x])
+    end
+  end
```
Should return an **enumerator**
```ruby
irb(main):007:0> array
=> [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
irb(main):008:0> array.each
=> #<Enumerator: [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]:each>
irb(main):009:0> array.my_each
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):9
        3: from enumerable.rb:5:in `my_each'
        2: from enumerable.rb:5:in `times'
        1: from enumerable.rb:6:in `block in my_each'
LocalJumpError (no block given (yield))
irb(main):010:0> 
```

maelfosso commented on enumerable.rb:
```diff
@@ -0,0 +1,75 @@
+    length.times do |x|
+      yield(x, self[x])
+    end
+  end
```
```ruby
irb(main):013:0> array
=> [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
irb(main):014:0> my_each_output = ''
=> ""
irb(main):015:0>       block = proc { |num, idx| my_each_output += "Num: #{num}, idx: #{idx}\n" }
=> #<Proc:0x000055ea504bc7e8@(irb):15>
irb(main):016:0>       array.each_with_index(&block)
=> [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
irb(main):017:0>       each_output = my_each_output.dup
=> "Num: 7, idx: 0\nNum: 8, idx: 1\nNum: 1, idx: 2\nNum: 4, idx: 3\nNum: 5, idx: 4\nNum: 0, idx: 5\nNum: 8, idx: 6\nNum: 4, idx: 7\nNum: 7, idx: 8\nNum: 6, idx: 9\nNum: 7, idx: 10\nNum: 8, idx: 11\nNum: 4, idx: 12\nNum: 8, idx: 13\nNum: 8, idx: 14\nNum: 6, idx: 15\nNum: 8, idx: 16\nNum: 0, idx: 17\nNum: 1, idx: 18\nNum: 7, idx: 19\nNum: 0, idx: 20\nNum: 2, idx: 21\nNum: 6, idx: 22\nNum: 6, idx: 23\nNum: 3, idx: 24\nNum: 1, idx: 25\nNum: 6, idx: 26\nNum: 6, idx: 27\nNum: 8, idx: 28\nNum: 5, idx: 29\nNum: 0, idx: 30\nNum: 2, idx: 31\nNum: 3, idx: 32\nNum: 7, idx: 33\nNum: 8, idx: 34\nNum: 7, idx: 35\nNum: 1, idx: 36\nNum: 5, idx: 37\nNum: 4, idx: 38\nNum: 4, idx: 39\nNum: 5, idx: 40\nNum: 1, idx: 41\nNum: 3, idx: 42\nNum: 5, idx: 43\nNum: 8, idx: 44\nNum: 7, idx: 45\nNum: 3, idx: 46\nNum: 3, idx: 47\nNum: 0, idx: 48\nNum: 0, idx: 49\nNum: 4, idx: 50\nNum: 1, idx: 51\nNum: 6, idx: 52\nNum: 2, idx: 53\nNum: 0, idx: 54\nNum: 4, idx: 55\nNum: 2, idx: 56\nNum: 1, idx: 57\nNum: 8, idx: 58\nNum: 3, idx: 59\nNum: 3, idx: 60\nNum: 6, idx: 61\nNum: 0, idx: 62\nNum: 8, idx: 63\nNum: 4, idx: 64\nNum: 4, idx: 65\nNum: 1, idx: 66\nNum: 6, idx: 67\nNum: 4, idx: 68\nNum: 6, idx: 69\nNum: 5, idx: 70\nNum: 3, idx: 71\nNum: 6, idx: 72\nNum: 6, idx: 73\nNum: 8, idx: 74\nNum: 6, idx: 75\nNum: 8, idx: 76\nNum: 7, idx: 77\nNum: 0, idx: 78\nNum: 6, idx: 79\nNum: 6, idx: 80\nNum: 2, idx: 81\nNum: 8, idx: 82\nNum: 8, idx: 83\nNum: 2, idx: 84\nNum: 1, idx: 85\nNum: 8, idx: 86\nNum: 5, idx: 87\nNum: 0, idx: 88\nNum: 5, idx: 89\nNum: 0, idx: 90\nNum: 3, idx: 91\nNum: 6, idx: 92\nNum: 4, idx: 93\nNum: 5, idx: 94\nNum: 8, idx: 95\nNum: 3, idx: 96\nNum: 3, idx: 97\nNum: 2, idx: 98\nNum: 1, idx: 99\n"
irb(main):018:0>      irb(main):013:0> array
=> [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
irb(main):014:0> my_each_output = ''
=> ""
irb(main):015:0> block = proc { |num, idx| my_each_output += "Num: #{num}, idx: #{idx}\n" }
=> #<Proc:0x000055ea504bc7e8@(irb):15>
irb(main):016:0> array.each_with_index(&block)
=> [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
irb(main):017:0> each_output = my_each_output.dup
=> "Num: 7, idx: 0\nNum: 8, idx: 1\nNum: 1, idx: 2\nNum: 4, idx: 3\nNum: 5, idx: 4\nNum: 0, idx: 5\nNum: 8, idx: 6\nNum: 4, idx: 7\nNum: 7, idx: 8\nNum: 6, idx: 9\nNum: 7, idx: 10\nNum: 8, idx: 11\nNum: 4, idx: 12\nNum: 8, idx: 13\nNum: 8, idx: 14\nNum: 6, idx: 15\nNum: 8, idx: 16\nNum: 0, idx: 17\nNum: 1, idx: 18\nNum: 7, idx: 19\nNum: 0, idx: 20\nNum: 2, idx: 21\nNum: 6, idx: 22\nNum: 6, idx: 23\nNum: 3, idx: 24\nNum: 1, idx: 25\nNum: 6, idx: 26\nNum: 6, idx: 27\nNum: 8, idx: 28\nNum: 5, idx: 29\nNum: 0, idx: 30\nNum: 2, idx: 31\nNum: 3, idx: 32\nNum: 7, idx: 33\nNum: 8, idx: 34\nNum: 7, idx: 35\nNum: 1, idx: 36\nNum: 5, idx: 37\nNum: 4, idx: 38\nNum: 4, idx: 39\nNum: 5, idx: 40\nNum: 1, idx: 41\nNum: 3, idx: 42\nNum: 5, idx: 43\nNum: 8, idx: 44\nNum: 7, idx: 45\nNum: 3, idx: 46\nNum: 3, idx: 47\nNum: 0, idx: 48\nNum: 0, idx: 49\nNum: 4, idx: 50\nNum: 1, idx: 51\nNum: 6, idx: 52\nNum: 2, idx: 53\nNum: 0, idx: 54\nNum: 4, idx: 55\nNum: 2, idx: 56\nNum: 1, idx: 57\nNum: 8, idx: 58\nNum: 3, idx: 59\nNum: 3, idx: 60\nNum: 6, idx: 61\nNum: 0, idx: 62\nNum: 8, idx: 63\nNum: 4, idx: 64\nNum: 4, idx: 65\nNum: 1, idx: 66\nNum: 6, idx: 67\nNum: 4, idx: 68\nNum: 6, idx: 69\nNum: 5, idx: 70\nNum: 3, idx: 71\nNum: 6, idx: 72\nNum: 6, idx: 73\nNum: 8, idx: 74\nNum: 6, idx: 75\nNum: 8, idx: 76\nNum: 7, idx: 77\nNum: 0, idx: 78\nNum: 6, idx: 79\nNum: 6, idx: 80\nNum: 2, idx: 81\nNum: 8, idx: 82\nNum: 8, idx: 83\nNum: 2, idx: 84\nNum: 1, idx: 85\nNum: 8, idx: 86\nNum: 5, idx: 87\nNum: 0, idx: 88\nNum: 5, idx: 89\nNum: 0, idx: 90\nNum: 3, idx: 91\nNum: 6, idx: 92\nNum: 4, idx: 93\nNum: 5, idx: 94\nNum: 8, idx: 95\nNum: 3, idx: 96\nNum: 3, idx: 97\nNum: 2, idx: 98\nNum: 1, idx: 99\n"
irb(main):018:0> my_each_output = ''
=> ""
irb(main):019:0> array.my_each_with_index(&block)
=> 100
```

maelfosso replied:
```ruby
irb(main):026:0> array
=> [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
irb(main):027:0> array.each_with_index
=> #<Enumerator: [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]:each_with_index>
irb(main):028:0> array.my_each_with_index
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):28
        3: from enumerable.rb:11:in `my_each_with_index'
        2: from enumerable.rb:11:in `times'
        1: from enumerable.rb:12:in `block in my_each_with_index'
LocalJumpError (no block given (yield))
irb(main):029:0>
```

maelfosso commented on enumerable.rb:
```diff
@@ -0,0 +1,75 @@
+      item.push(self[x]) if yield(self[x])
+    end
+    item
+  end
```
Should return an **enumerator**
```ruby
irb(main):031:0> array
=> [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
irb(main):032:0> array.select
=> #<Enumerator: [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]:select>
irb(main):033:0> array.my_select
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):33
        3: from enumerable.rb:18:in `my_select'
        2: from enumerable.rb:18:in `times'
        1: from enumerable.rb:19:in `block in my_select'
LocalJumpError (no block given (yield))
irb(main):034:0>
```

maelfosso commented on enumerable.rb:
```diff
@@ -0,0 +1,75 @@
+      value = false unless yield(self[x])
+    end
+    value
+  end
```
```ruby
irb(main):054:0> true_array = [1, true, 'hi', []]
=> [1, true, "hi", []]
irb(main):055:0> true_array.all?
=> true
irb(main):056:0> true_array.my_all
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):56
        3: from enumerable.rb:26:in `my_all'
        2: from enumerable.rb:26:in `times'
        1: from enumerable.rb:27:in `block in my_all'
LocalJumpError (no block given (yield))
irb(main):057:0>
```

maelfosso replied:
```ruby
irb(main):059:0> array
=> [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
irb(main):060:0> array.all?(Integer)
=> true
irb(main):061:0> array.my_all(Integer)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):61
        1: from enumerable.rb:24:in `my_all'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):062:0> 
```

maelfosso replied:
```ruby
irb(main):065:0> words = %w[dog door rod blade]
=> ["dog", "door", "rod", "blade"]
irb(main):066:0> words.all?(/d/)
=> true
irb(main):067:0> words.my_all(/d/)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):67
        1: from enumerable.rb:24:in `my_all'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):068:0> 
```

maelfosso replied:
```ruby
irb(main):070:0> array
=> [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
irb(main):071:0> array.all?(3)
=> false
irb(main):072:0> array.my_all(3)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):72
        1: from enumerable.rb:24:in `my_all'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):073:0> 
```

maelfosso commented on enumerable.rb:
```diff
@@ -0,0 +1,75 @@
+      value = true if yield(self[x])
+    end
+    value
+  end
```
```ruby
irb(main):075:0> true_block = proc { |num| num <= 3 }
=> #<Proc:0x000055ea507045a0@(irb):75>
irb(main):077:0> array.my_any(&true_blocK)
Traceback (most recent call last):
        2: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        1: from (irb):77
NameError (undefined local variable or method `true_blocK' for main:Object)
irb(main):078:0> array.any?(&true_block)
=> true
irb(main):079:0> array
=> [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
irb(main):080:0> 
```

maelfosso replied:
```ruby
irb(main):082:0> true_array = [nil, false, true, []]
=> [nil, false, true, []]
irb(main):083:0> true_array.any?
=> true
irb(main):084:0> true_array.my_any
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):84
        3: from enumerable.rb:34:in `my_any'
        2: from enumerable.rb:34:in `times'
        1: from enumerable.rb:35:in `block in my_any'
LocalJumpError (no block given (yield))
irb(main):085:0>
```

maelfosso replied:
```ruby
irb(main):087:0> array
=> [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
irb(main):088:0> array.any?(Integer)
=> true
irb(main):089:0> array.my_any(Integer)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):89
        1: from enumerable.rb:32:in `my_any'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):090:0>
```

maelfosso replied:
```ruby
irb(main):092:0> words
=> ["dog", "door", "rod", "blade"]
irb(main):093:0> words.any?(/z/)
=> false
irb(main):094:0> words.my_any(/z/)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):94
        1: from enumerable.rb:32:in `my_any'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):095:0> 
```

maelfosso replied:
```ruby
irb(main):097:0> words[0] = 'cat'
=> "cat"
irb(main):098:0> words.any?('cat')
=> true
irb(main):099:0> words.my_any('cat')
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):99
        1: from enumerable.rb:32:in `my_any'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):100:0> 
```

maelfosso commented on enumerable.rb:
```diff
@@ -0,0 +1,75 @@
+      return false if yield(self[x])
+    end
+    true
+  end
```
```ruby
irb(main):108:0> false_array = [nil, false, nil, false]
=> [nil, false, nil, false]
irb(main):109:0> false_array.none?
=> true
irb(main):110:0> false_array.my_none
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):110
        3: from enumerable.rb:41:in `my_none'
        2: from enumerable.rb:41:in `times'
        1: from enumerable.rb:42:in `block in my_none'
LocalJumpError (no block given (yield))
irb(main):111:0> 
```

maelfosso replied:
```ruby
irb(main):113:0> array
=> [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
irb(main):114:0> array.none?(String)
=> true
irb(main):115:0> array.my_none(String)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):115
        1: from enumerable.rb:40:in `my_none'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):116:0>
```

maelfosso replied:
```ruby
irb(main):121:0> words = %w[dog door rod blade]
=> ["dog", "door", "rod", "blade"]
irb(main):122:0> words.none?(/z/)
=> true
irb(main):123:0> words.my_none(/z/)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):123
        1: from enumerable.rb:40:in `my_none'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):124:0>
```

maelfosso replied:
```ruby
irb(main):126:0> words[0] = 5
=> 5
irb(main):127:0> words.none?(5)
=> false
irb(main):128:0> words.my_none(5)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):128
        1: from enumerable.rb:40:in `my_none'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):129:0> 
```

maelfosso commented on enumerable.rb:
```diff
@@ -0,0 +1,75 @@
+      item += 1 if yield(self[x])
+    end
+    item
+  end
```
```ruby
irb(main):131:0> array
=> [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
irb(main):132:0> array.count
=> 100
irb(main):133:0> array.my_count
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):133
        3: from enumerable.rb:49:in `my_count'
        2: from enumerable.rb:49:in `times'
        1: from enumerable.rb:50:in `block in my_count'
LocalJumpError (no block given (yield))
irb(main):134:0> 
```

maelfosso replied:
```ruby
irb(main):136:0> array
=> [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
irb(main):137:0> array.count(3)
=> 11
irb(main):138:0> array.my_count(3)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):138
        1: from enumerable.rb:47:in `my_count'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):139:0> 
```

maelfosso commented on enumerable.rb:
```diff
@@ -0,0 +1,75 @@
+      item.push(result)
+    end
+    item
+  end
```
```ruby
irb(main):141:0> array
=> [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
irb(main):142:0> array.map
=> #<Enumerator: [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]:map>
irb(main):143:0> array.my_map
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):143
        3: from enumerable.rb:57:in `my_map'
        2: from enumerable.rb:57:in `times'
        1: from enumerable.rb:58:in `block in my_map'
LocalJumpError (no block given (yield))
irb(main):144:0> 
```

maelfosso commented on enumerable.rb:
```diff
@@ -0,0 +1,75 @@
+      sum = yield(sum, self[x + 1])
+    end
+    sum
+  end
```
```ruby
irb(main):146:0> range = Range.new(5, 50)
=> 5..50
irb(main):147:0> range.my_inject(4) { |prod, n| prod * n }
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):147
        1: from enumerable.rb:64:in `my_inject'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):148:0> range.inject(4) { |prod, n| prod * n }
=> 5069015533618896340602101361010794807396273594826752000000000000
irb(main):149:0> 
```

maelfosso replied:
```ruby
irb(main):151:0> array
=> [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
irb(main):152:0> array.inject(:+)
=> 434
irb(main):153:0> array.my_inject(:+)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):153
        1: from enumerable.rb:64:in `my_inject'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):154:0>
```

maelfosso reviewed (changes requested):
Hello @men32z 

Thank you for the job you did. However there are bugs to fix.
Please, read the comments below to learn more about those errors.

Kindly, make the changes and submit a new code review request.
Thanks

men32z committed:
fix for the methods

men32z committed:
Merge branch 'enumerables' of github.com:men32z/advanced-building-blocks-enumerables into enumerables

men32z committed:
some fixes

men32z committed:
fixin sticlker errors

men32z committed:
test stickler fix

men32z committed:
fixes sticlker

men32z committed:
stickler 2

men32z committed:
stickler 3

samgaco commented on enumerable.rb:
```diff
@@ -0,0 +1,141 @@
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    length.times do |x|
```
- [x] my_each_with_index calls the given block once for each element in self

samgaco commented on enumerable.rb:
```diff
@@ -0,0 +1,141 @@
+  def my_all(field = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    return true if !block_given? && field.nil?
+
+    value = true
```
- [x] my_all? returns true if the block never returns false or nil
- [x] my_all? when no block or argument is given returns true when none of the collection members are false or nil
- [x] my_all? when a class is passed as an argument returns true if all of the collection is a member of such class
- [x] my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
- [x] my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

samgaco commented on enumerable.rb:
```diff
@@ -0,0 +1,141 @@
+  def my_any(field = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    return true if !block_given? && field.nil?
+
+    value = false
```
- [ ] my_any? returns true if the block ever returns a value other than false or nil
- [ ] my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
- [ ] my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
- [ ] my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [ ] my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

samgaco commented on enumerable.rb:
```diff
@@ -0,0 +1,141 @@
+  def my_none(field = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    return true if !block_given? && field.nil?
+
+    length.times do |x|
```
- [ ] my_none? returns true if the block never returns true for all elements
- [ ] my_none? when no block or argument is given returns true only if none of the collection members is true
- [ ] my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
- [ ] my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex
- [ ] my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern

samgaco commented on enumerable.rb:
```diff
@@ -0,0 +1,141 @@
+  def my_inject(field = nil)
+    arr = to_a
+    if field&.respond_to?(:is_a?) && field&.is_a?(Integer)
+      sum = field
```
- [ ] my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method

samgaco reviewed (changes requested):
Hi @men32z :high_brightness: 

Your methods still are not fully functional. You can find [here](https://ruby-doc.org/core-2.6.4/Enumerable.html) how are they expected to work, in the comments below you have the exact cases for which you still have to make some improvements. I hope it helps.

Regards.

---------------------------
moving on :arrow_down: 
* **[For your next code review requests]**  Add a short summary of changes in Pull Request description


---------------------------


* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

men32z committed:
my_each_with_index parameteres order fixed

men32z commented:
> 
> 
> Hi @men32z 🔆
> 
> Your methods still are not fully functional. You can find [here](https://ruby-doc.org/core-2.6.4/Enumerable.html) how are they expected to work, in the comments below you have the exact cases for which you still have to make some improvements. I hope it helps.
> 
> Regards.
> 
> moving on ⬇️
> 
>     * **[For your next code review requests]**  Add a short summary of changes in Pull Request description
> 
> 
>     * ###### STATUS OF THE PROJECT: REQUESTED CHANGES 🛑

@samgaco I see all methods working right, i did this file to test with the help of the previous review
`
#my_all
puts "#my_all"
puts "-----------------------------------------------------------------------------"

true_array = [1, true, 'hi', []]
puts true_array.all?
puts true_array.my_all

array = [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
puts array.all?(Integer)
puts array.my_all(Integer)

words = %w[dog door rod blade]
puts words.all?(/d/)
puts words.my_all(/d/)

puts array.all?(3)
puts array.my_all(3)

#my_any
puts "#my_any"
puts "-----------------------------------------------------------------------------"

array = [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
true_block = proc { |num| num <= 3 }
puts array.any?(&true_block)
puts array.my_any(&true_block)

true_array = [nil, false, true, []]
puts true_array.any?
puts true_array.my_any

puts array.any?(Integer)
puts array.my_any(Integer)

words = ["dog", "door", "rod", "blade"]
puts words.any?(/z/)
puts words.my_any(/z/)

words[0] = 'cat'
puts words.any?('cat')
puts words.my_any('cat')

#my_none
puts "#my_none"
puts "-----------------------------------------------------------------------------"

false_array = [nil, false, nil, false]

puts false_array.none?
puts false_array.my_none

array = [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
puts array.none?(String)
puts array.my_none(String)

words = %w[dog door rod blade]
puts words.none?(/z/)
puts words.my_none(/z/)

words[0] = 5
puts words.none?(5)
puts words.my_none(5)

#my_count
puts "#my_count"
puts "-----------------------------------------------------------------------------"

array = [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
puts array.count
puts array.my_count

puts array.count(3)
puts array.my_count(3)

puts array.count { |x| x%2 == 0 }
puts array.my_count { |x| x%2 == 0 }

#my_map
puts "#my_map"
puts "-----------------------------------------------------------------------------"

array = [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
puts array.map
puts array.my_map

print array.map { |n| n * 2 }
print array.my_map { |n| n * 2 }

#my_inject
puts "#my_inject"
puts "-----------------------------------------------------------------------------"

range = Range.new(5, 50)

puts range.inject(4) { |prod, n| prod * n }
puts range.my_inject(4) { |prod, n| prod * n }

array = [7, 8, 1, 4, 5, 0, 8, 4, 7, 6, 7, 8, 4, 8, 8, 6, 8, 0, 1, 7, 0, 2, 6, 6, 3, 1, 6, 6, 8, 5, 0, 2, 3, 7, 8, 7, 1, 5, 4, 4, 5, 1, 3, 5, 8, 7, 3, 3, 0, 0, 4, 1, 6, 2, 0, 4, 2, 1, 8, 3, 3, 6, 0, 8, 4, 4, 1, 6, 4, 6, 5, 3, 6, 6, 8, 6, 8, 7, 0, 6, 6, 2, 8, 8, 2, 1, 8, 5, 0, 5, 0, 3, 6, 4, 5, 8, 3, 3, 2, 1]
puts array.inject(:+)
puts array.my_inject(:+)


#my_each_with_index
puts "#my_inject"
puts "-----------------------------------------------------------------------------"

hash = Hash.new
%w(cat dog wombat).each_with_index { |item, index|
  hash[item] = index
}

puts hash

hash = Hash.new
%w(cat dog wombat).my_each_with_index { |item, index|
  hash[item] = index
}

puts hash`

this are the results.

$ ruby enumerable.rb
#my_all
-----------------------------------------------------------------------------
true
true
true
true
true
true
false
false
#my_any
-----------------------------------------------------------------------------
true
true
true
true
true
true
false
false
true
true
#my_none
-----------------------------------------------------------------------------
true
true
true
true
true
true
false
false
#my_count
-----------------------------------------------------------------------------
100
100
11
11
62
62
#my_map
-----------------------------------------------------------------------------
#<Enumerator:0x0000000005141478>
#<Enumerator:0x0000000005141360>
[14, 16, 2, 8, 10, 0, 16, 8, 14, 12, 14, 16, 8, 16, 16, 12, 16, 0, 2, 14, 0, 4, 12, 12, 6, 2, 12, 12, 16, 10, 0, 4, 6, 14, 16, 14, 2, 10, 8, 8, 10, 2, 6, 10, 16, 14, 6, 6, 0, 0, 8, 2, 12, 4, 0, 8, 4, 2, 16, 6, 6, 12, 0, 16, 8, 8, 2, 12, 8, 12, 10, 6, 12, 12, 16, 12, 16, 14, 0, 12, 12, 4, 16, 16, 4, 2, 16, 10, 0, 10, 0, 6, 12, 8, 10, 16, 6, 6, 4, 2][14, 16, 2, 8, 10, 0, 16, 8, 14, 12, 14, 16, 8, 16, 16, 12, 16, 0, 2, 14, 0, 4, 12, 12, 6, 2, 12, 12, 16, 10, 0, 4, 6, 14, 16, 14, 2, 10, 8, 8, 10, 2, 6, 10, 16, 14, 6, 6, 0, 0, 8, 2, 12, 4, 0, 8, 4, 2, 16, 6, 6, 12, 0, 16, 8, 8, 2, 12, 8, 12, 10, 6, 12, 12, 16, 12, 16, 14, 0, 12, 12, 4, 16, 16, 4, 2, 16, 10, 0, 10, 0, 6, 12, 8, 10, 16, 6, 6, 4, 2]#my_inject
-----------------------------------------------------------------------------
5069015533618896340602101361010794807396273594826752000000000000
5069015533618896340602101361010794807396273594826752000000000000
434
434
#my_inject
-----------------------------------------------------------------------------
{"cat"=>0, "dog"=>1, "wombat"=>2}
{"cat"=>0, "dog"=>1, "wombat"=>2}

maelfosso commented on enumerable.rb:
```diff
@@ -0,0 +1,141 @@
+      end
+    end
+    value
+  end
```
Sorry there is a bug here 
```ruby
irb(main):006:0> false_array = [1, false, 'hi', []]
=> [1, false, "hi", []]
irb(main):007:0> false_array.all?
=> false
irb(main):008:0> false_array.my_all
=> true
irb(main):009:0> 
```

maelfosso commented on enumerable.rb:
```diff
@@ -0,0 +1,141 @@
+      end
+    end
+    value
+  end
```
There is still bugs here 
```ruby
irb(main):011:0> false_array = [nil, false, nil, false]
=> [nil, false, nil, false]
irb(main):012:0> false_array.any?
=> false
irb(main):013:0> false_array.my_any
=> true
irb(main):014:0>
```

maelfosso commented on enumerable.rb:
```diff
@@ -0,0 +1,141 @@
+      end
+    end
+    true
+  end
```
There is still bugs here
```ruby
irb(main):016:0> true_array = [nil, false, true, []]
=> [nil, false, true, []]
irb(main):017:0> true_array.none?
=> false
irb(main):018:0> true_array.my_none
=> true
irb(main):019:0> 
```

maelfosso commented on enumerable.rb:
```diff
@@ -0,0 +1,141 @@
+      sum = o ? sum.method(o).call(arr[x + d_m]) : sum = yield(sum, arr[x + d_m])
+    end
+    sum
+  end
```
Sorry there is still bug here
```ruby
irb(main):021:0> range = Range.new(5, 50)
=> 5..50
irb(main):022:0> range.inject(2, :*)
=> 2534507766809448170301050680505397403698136797413376000000000000
irb(main):023:0> range.my_inject(2, :*)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):23
        1: from enumerable.rb:115:in `my_inject'
ArgumentError (wrong number of arguments (given 2, expected 0..1))
irb(main):024:0>
```

maelfosso reviewed (changes requested):
Hello @samgaco 

Thank for the improvements you did. 
You're about to closed the project. There is just 4 bugs to fix.
Read the comment left below to know more about the bugs.

Please, fix them and submit a new code review request.
**Status: Not approved**

Thanks

men32z committed:
fixed errors

- method my_all now check each element if there is not block and parameter.
- method my_any now check each element if there is not block and parameter.
- method my_none now check each element if there is not block and parameter.
- method my_inject can recieve 2 parameters.

BalintMendli reviewed (approved):
Hi @men32z ,

Great job. 👏 

All methods work fine now.

It is a 👍 from me.

You can go ahead and merge the project.
